ksqldb-engine/src/main/java/io/confluent/ksql/planner/LogicalPlanner.java

```diff
@@ -872,21 +883,33 @@ private static boolean isInnerNode(final PlanNode node) {
     throw new IllegalStateException("Unknown node type: " + node.getClass().getName());
   }
 
-  private JoinKey buildForeignJoinKey(final Join join, final ColumnReferenceExp columnRef) {
+  private JoinKey buildForeignJoinKey(final Join join,
+                                      final Expression foreignKeyExpression) {
     final AliasedDataSource leftSource = join.getInfo().getLeftSource();
     final SourceName alias = leftSource.getAlias();
     final List<QualifiedColumnReferenceExp> leftSourceKeys =
         leftSource.getDataSource().getSchema().key().stream()
             .map(c -> new QualifiedColumnReferenceExp(alias, c.name()))
             .collect(Collectors.toList());
 
-    final ColumnName foreignKeyColumnName = columnRef.maybeQualifier().isPresent()
-        ? ColumnNames.generatedJoinColumnAlias(
-            columnRef.maybeQualifier().get(),
-            columnRef.getColumnName())
-        : columnRef.getColumnName();
+    final VisitParentExpressionVisitor<Optional<Expression>, Context<Void>> aliasRewritter =
```

Comment:
Similar to the code we replace, we need to rewrite the expression with the alias prefix...